tests/components/coolmaster/conftest.py

```diff
@@ -0,0 +1,95 @@
+"""Fixtures for the Coolmaster integration."""
+from __future__ import annotations
+
+from typing import Any
+from unittest.mock import patch
+
+import pytest
+
+from homeassistant.components.coolmaster.const import DOMAIN
+from homeassistant.core import HomeAssistant
+
+from tests.common import MockConfigEntry
+
+DEFAULT_INFO: dict[str, str] = {
+    "version": "1",
+}
+
+DEFUALT_UNIT_DATA: dict[str, Any] = {
+    "is_on": False,
+    "thermostat": 20,
+    "temperature": 25,
+    "fan_speed": "low",
+    "mode": "cool",
+    "error_code": None,
+    "clean_filter": False,
+    "swing": None,
+    "temperature_unit": "celsius",
+}
+
+TEST_UNITS: dict[dict[str, Any]] = {
+    "L1.100": {**DEFUALT_UNIT_DATA},
+    "L1.101": {
+        **DEFUALT_UNIT_DATA,
+        **{
+            "is_on": True,
+            "clean_filter": True,
+            "error_code": "Err1",
+        },
+    },
+}
+
+
+class CoolMasterNetUnitMock:
+    """Mock for CoolMasterNetUnit."""
+
+    def __init__(self, unit_id: str, attributes: dict[str, Any]) -> None:
+        """Initialize the CoolMasterNetUnitMock."""
+        self.unit_id = unit_id
+        for key, value in attributes.items():
+            setattr(self, key, value)
+
+    async def reset_filter(self):
+        """Report that the air filter was cleaned and reset the timer."""
+        TEST_UNITS[self.unit_id]["clean_filter"] = False
```

Comment:
I was about read to merge and doing a final pass, but then just noticed this is mutating global state, which is going to be a problem. You're going to want to make some kind of data object (or single instance can be shared across the mocks)

Reply:
Fixed. Thanks!